bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/config/ConfigDescriptionResource.java

```diff
@@ -0,0 +1,96 @@
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionDTO;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionDTOMapper;
+import org.eclipse.smarthome.io.rest.LocaleUtil;
+import org.eclipse.smarthome.io.rest.RESTResource;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * {@link ConfigDescriptionResource} provides access to {@link ConfigDescription}s via REST.
+ *
+ * @author Dennis Nobel - Initial contribution
+ */
+@Path(ConfigDescriptionResource.PATH_CONFIG_DESCRIPTIONS)
+@Api
+public class ConfigDescriptionResource implements RESTResource {
+
+    /** The URI path to this resource */
+    public static final String PATH_CONFIG_DESCRIPTIONS = "config-descriptions";
+
+    private final class ConfigDescriptionConverter implements Function<ConfigDescription, ConfigDescriptionDTO> {
+        @Override
+        public ConfigDescriptionDTO apply(ConfigDescription configDescription) {
+            return ConfigDescriptionDTOMapper.map(configDescription);
+        }
+    }
+
+    private ConfigDescriptionRegistry configDescriptionRegistry;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Gets all available config descriptions.", response = ConfigDescriptionDTO.class, responseContainer = "List")
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK") )
+    public Response getAll(@HeaderParam("Accept-Language") @ApiParam(value = "Accept-Language") String language) {
+        Locale locale = LocaleUtil.getLocale(language);
+        Iterable<ConfigDescriptionDTO> transform = Iterables
+                .transform(configDescriptionRegistry.getConfigDescriptions(locale), new ConfigDescriptionConverter());
+        return Response.ok(Lists.newArrayList(transform)).build();
+    }
+
+    @GET
+    @Path("/{uri}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Gets a config description by URI.", response = ConfigDescriptionDTO.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 400, message = "Invalid URI syntax"), @ApiResponse(code = 404, message = "Not found") })
+    public Response getByURI(@HeaderParam("Accept-Language") @ApiParam(value = "Accept-Language") String language,
+            @PathParam("uri") @ApiParam(value = "uri") String uri) {
+        Locale locale = LocaleUtil.getLocale(language);
+        try {
+            ConfigDescription configDescription = this.configDescriptionRegistry.getConfigDescription(new URI(uri),
+                    locale);
+            return configDescription != null ? Response.ok(configDescription).build() : Response.status(404).build();
```

Comment:
An additional remark: Serialization of the original classes works directly, but not the deserialization, because GSON does not know to to turn a string into an URI object. For the moment config descriptions are read-only, but if we ever decide to make them editable or we wend events for it, we will also need deserialization and therefore the DTO objects.